batch/batches/CaptureThumb/KAsyncCaptureThumb.class.php

```diff
@@ -211,12 +170,246 @@ private function getMediaInfoData($partnerId, $srcAssetId)
+
+		$data->thumbPath = $thumbPath;
+		$job = $this->moveFile($job, $data);
+
+		if($this->checkFileExists($job->data->thumbPath))
+		{
+			$updateData = new KalturaCaptureThumbJobData();
+			$updateData->thumbPath = $data->thumbPath;
+			return $this->closeJob($job, null, null, null, KalturaBatchJobStatus::FINISHED, $updateData);
+		}
+
+		return $this->closeJob($job, KalturaBatchJobErrorTypes::APP, KalturaBatchJobAppErrors::NFS_FILE_DOESNT_EXIST, 'File not moved correctly', KalturaBatchJobStatus::FAILED, $data);
+	}
+
+	protected function createBif($job, $data, $rootPath, $mediaFile, $thumbParamsOutput)
+	{
+		$capturePath = null;
+		$images = array();
+		if($data->srcAssetType == KalturaAssetType::FLAVOR)
+		{
+			list($mediaInfoWidth, $mediaInfoHeight, $mediaInfoDar, $mediaInfoVidDur, $mediaInfoScanType, $mediaInfoVideoRotation) = $this->getMediaInfoData($job->partnerId, $data->srcAssetId);
+			$bifInterval = $this->getBifInterval($thumbParamsOutput);
+
+			$captureVidSec = $bifInterval;
+			$folderPath = realpath($rootPath) . DIRECTORY_SEPARATOR . uniqid();
+			$rootPath = self::createDir($folderPath);
+			if (!$rootPath)
+				die();
+
+			$generalCapturePath = $this->createUniqFileName($rootPath);
+
+			$count = floor($mediaInfoVidDur / $bifInterval);
+			KalturaLog::debug("Number of images to capture - [$count]");
+
+			$packagerRetries = 3;
+			$this->impersonate($job->partnerId);
+			$entry = self::$kClient->baseEntry->get($job->entryId);
+			$srcFlavorAsset = self::$kClient->flavorAsset->get($data->srcAssetId);
+			$this->unimpersonate();
+			if(!$entry || !$srcFlavorAsset)
+			{
+				KalturaLog::debug("Could not get entry - [$job->entryId] or flavor [$srcFlavorAsset]. Capturing using ffmpeg.");
+				$packagerRetries = 0;
+			}
+
+			$params = array($thumbParamsOutput->density, $thumbParamsOutput->quality, $mediaInfoVideoRotation, $thumbParamsOutput->cropX, $thumbParamsOutput->cropY,
+				$thumbParamsOutput->cropWidth, $thumbParamsOutput->cropHeight, $thumbParamsOutput->stripProfiles);
+			// 5 - always force setting the given dimensions
+			$shouldResizeByPackager = KThumbnailCapture::shouldResizeByPackager($params, 5, array($thumbParamsOutput->width, $thumbParamsOutput->height));
+			list($picWidth, $picHeight) = $shouldResizeByPackager ? array($thumbParamsOutput->width, $thumbParamsOutput->height) : array(null, null);
+
+			while($count--)
+			{
+				$capturePath = $generalCapturePath . '_sec_' . $captureVidSec . '_';
+				if($packagerRetries)
+				{
+					$thumbCaptureByPackager = false;
+					$success = KThumbnailCapture::captureLocalThumbForBifUsingPackager($mediaFile, $capturePath, $captureVidSec, $picWidth, $picHeight);
+					$packagerResizeFullPath = $capturePath . self::TEMP_FILE_POSTFIX;
+					KalturaLog::debug("Packager capture is [$success] with dimension [$picWidth,$picHeight] and packagerResize [$shouldResizeByPackager] in path [$packagerResizeFullPath]");
+					if(!$success)
+					{
+						$packagerRetries--;
```

Comment:
Add Warning to the batch log that we stopped using packer due to errors from packager